util/file.go

```diff
@@ -15,6 +15,30 @@ func FileExists(path string) bool {
 	return true
 }
 
+func IsDirectory(path string) bool {
+	stat, err := os.Stat(path)
+	if err != nil {
+		return false
+	}
+	return stat.IsDir()
+}
+
+func ListDirectory(path string) []string {
+	var fileNames []string
+
+	err := filepath.Walk(path, func (path string, info os.FileInfo, err error) error {
+		if ! info.IsDir() {
+			fileNames = append(fileNames, path)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil
```

Comment:
It's wrong. When err is not nil, we should return err like below.

```go
return fileNames, err
```

Reply:
Done. Instead of returning `fileNames, err`, it's now returning `nil, err`, and the caller exits in case of error; I think it's a better behavior than continuing if it failed to resolve some of the file names.